### PR TITLE
GEOMESA-2537 fix:FileMetadataTest will not test correctly

### PR DESCRIPTION
**Pull request by misslzyan**
add reload for clear the cache when do test cases . 
Signed-off-by: 段卫东 <misslzyan108@gmail.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
@misslzyan what is this trying to fix? it seems like the tests pass ok without the change. Thanks,

**Comment by misslzyan** [comment]
@elahrvivaz   The tests with persist and load , the load will use cache but not the metadata which was persisted .  This case will cause the tests pass but not checks the code correctly.
eg.
data.writeUTF(str)   
the test case 'create and persist an empty metadata file' will pass through , but this line code is error .

**Comment by elahrvivaz** [comment]
ah, I see. makes sense, thanks!

File: geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/StorageMetadata.scala [hunk 1]
```diff
@@ -227,7 +227,16 @@ object StorageMetadata extends MethodProfiling with LazyLogging {
...
-  def load(fc: FileContext, root: Path): Option[StorageMetadata] = Option(cache.get((fc, root)))
+  def load(fc: FileContext, root: Path , reload: Boolean = false): Option[StorageMetadata] = {
+    if(reload) {
+      val metadata = loadStorage(fc, root)
+      metadata match {
+        case Some(v) => cache.put((fc, root), v)
+        case None    => cache.put((fc, root), null)
+      }
+      metadata
+    }else Option(cache.get((fc, root)))
```
Comment (elahrvivaz) [inline geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/StorageMetadata.scala]: could you clean up the formatting a bit here? there's an extra space before the comma on line 230, we try to put a space after an if, and we always try to use {} for an if/else. e.g.
```
if (reload) {
...
} else {
  ...
}
```

**Comment by elahrvivaz** [comment]
@misslzyan thanks for the fix! it looks good, if you don't mind just cleaning up the code formatting I will merge it.

**Comment by misslzyan** [comment]
@elahrvivaz   I will clean up the code soon

**Comment by misslzyan** [comment]
@elahrvivaz   Done ! I should go to sleep . You can leave words here ,and I will read them tomorrow.Thanks.

**Pull request merged**